script/validators/source_validator.py

```diff
@@ -127,19 +129,29 @@ def check_format(file_path):
         clang_format_cmd = [CLANG_FORMAT, "-style=file", file_path]
         formatted_src = subprocess.check_output(clang_format_cmd).splitlines(True)
         # Load source file
-        file = open(file_path, "r")
-        src = file.readlines()
+        with open(file_path, "r") as file:
+            src = file.readlines()
 
         # Do the diff
         diff = difflib.unified_diff(src, formatted_src)
-        for line in diff:
-            LOG.info("Invalid formatting in file : " + file_path)
+        if diff.next():  # Got a non-empty diff.
             status = False
+        
+        metadata_over = False
+        for line in diff:
+            if line.startswith('@@'):
+                metadata_over = True
+                continue
+            if not metadata_over:
+                continue
+            if line.startswith('-'):
+                LOG.info("Invalid formatting in file : " + file_path)
+                LOG.info("Line: %s" % line[1:])
```

Comment:
In my opinion the content of the line is not useful without line numbers. As we have a formatter script for that, you can leave it the old way. Or you find a way to display the line numbers here, I don't know if libdiff can do that.